common/error.h

```diff
@@ -20,7 +20,7 @@ namespace Carbon {
 struct Success {};
 
 // Tracks an error message.
-class [[nodiscard]] Error : public Printable<Error> {
```

Comment:
I definitely think we should keep the attribute here: when a function returns `Error`, there's a high risk of its return value being ignored, because it's not salient information on the happy path. Adding this attribute to `absl::Status` was a huge win for the Google codebase

Reply:
llvm::Error uses nodiscard so keeping it here for consistency, with a comment.